File: Scala/src/main/scala/com/mine/dao/jdbc/JDBCUtils.scala
package com.mine.dao.jdbc

import java.sql._

object JDBCUtils {

    def getConnection(driver: String, urlWithNameAndPassword: String): Connection = {
        try {
            Class.forName(driver)
            return DriverManager.getConnection(urlWithNameAndPassword)
        } catch {
            case e: ClassNotFoundException => e.printStackTrace()
            case e: SQLException => e.printStackTrace()
        }
        null
    }

    def close(c: Connection, s: Statement): Unit = {
        if (s != null) try s.close() catch {
            case e: SQLException => e.printStackTrace()
        }
        if (c != null) try c.close() catch {
            case e: SQLException => e.printStackTrace()
        }
    }

    def close(c: Connection, s: Statement, r: ResultSet): Unit = {
        if (r != null) try r.close() catch {
            case e: SQLException => e.printStackTrace()
        }
        if (s != null) try s.close() catch {
            case e: SQLException => e.printStackTrace()
        }
        if (c != null) try c.close() catch {
            case e: SQLException => e.printStackTrace()
        }
    }
}

File: ProjectUtils/src/main/java/cn/mine/util/JdbcUtils.java
package cn.mine.util;

import com.mchange.v2.c3p0.ComboPooledDataSource;

import javax.sql.DataSource;
import java.sql.Connection;
import java.sql.SQLException;

/**
 * @author 魏喜明 2021-02-03 00:32:07
 */
public class JdbcUtils {
    private static final DataSource dataSource;
    private static Connection connection;

    static {
        // 数据源只能被创建一次
        dataSource = new ComboPooledDataSource("mvc");
    }

    /**
     * 返回数据源的连接对象
     *
     * @return 返回数据源的连接对象
     */
    public static Connection getConnection() {
        try {
            connection = dataSource.getConnection();
        } catch (SQLException throwable) {
            throwable.printStackTrace();
        }
        return connection;
    }

    /**
     * 释放 Connection 连接
     *
     * @param connection Connection 对象
     */
    public static void releaseConnection(Connection connection) {
        if (connection != null) {
            try {
                connection.close();
            } catch (SQLException throwable) {
                throwable.printStackTrace();
            }
        }
    }
}
